app/services/uuid_reporter.rb

```diff
@@ -0,0 +1,134 @@
+class UuidReporter
+  def self.run(**opts)
+    new(**opts).run
+  end
+
+  # Instantiate the service object
+  #
+  # @attr email_file [String] the location of the file with the list of emails
+  # @attr sp_file [String] the location of the file with the list of SP issuers
+  # @attr output [String] the location where the output CSV should be written
+  def initialize(email_file:, sp_file:, output:)
+    @emails = parse_file(email_file).map(&:downcase).uniq
+    validate_emails
+    @issuers = parse_file(sp_file).uniq
+    validate_issuers
+    @output = output
+    validate_output
+  end
+
+  # Take the lists of ServiceProvider issuers and emails and do the following:
+  #   1. Confirm that the user has an Identity with at least one of the SPs
+  #   2. Find the user's associated AgencyIdentity and uuid
+  #   3. Export a CSV file with email addresses and associated uuids
+  #
+  # @return [Integer] the number of uuids collected
+  def run
+    @agency = find_agency
+    @user_ids = collect_user_ids
+    @identity_hash = collect_identities
+    save_csv
+
+    identity_hash.length
+  end
+
+  private
+
+  attr_reader :emails, :issuers, :output, :agency, :user_ids, :identity_hash
+
+  def parse_file(file)
+    File.readlines(file, chomp: true)
```

Comment:
omg TIL about the `chomp:` option!! I had been doing `.map(&:chomp)` like a chump